### ISSUE TITLE
NW1 and NW2: (0.44.0 nightly build) [REGRESSION] Restoring a maximized window is not consistent anymore

### ISSUE THREAD

**Issue by DimPaDev** [nwjs/nw.js#7341]
NWJS Version : 0.44.0 nightly build ( http://dl.nwjs.io/live-build/nw44/20200124-162000/0d1fb6ffb/v0.44.0/ )
Operating System : Linux Fedora 31 (KDE plasma and gnome-shell)

### Expected behavior

On Linux, When you restore ( nw.Window.get().restore() ) a maximized window ( nw.Window.get().maximize() ) the window sould be restored in the previous state with the original position and size.

### Actual behavior

In the last nightly build this feature (restoring a maximized window) acts weirdly , sometimes the window gets restored with the wrong size in another position. I tested the same code with NW1 mode and NW2 mode and with a native frame and without a frame but nothing changes, the bug is still there. The bug seems to be Linux-related only, I tested the demo on Windows 10 and it seems that it works just fine. 

### How to reproduce

I created a minimal demo where you can press the letter m to maximize the window and the letter r to restore it. If you press it several times you will notice that sooner or later the bug will appear.

**DOWNLOAD DEMO:**
[nw-demo--restore-bug.zip](https://github.com/nwjs/nw.js/files/4111466/nw-demo--restore-bug.zip)

**Comment by DimPaDev** [comment]
![ezgif-6-eef16800dbf9](https://user-images.githubusercontent.com/42065117/73113865-76039c80-3f16-11ea-843d-159eb7adf6ec.gif)

**Comment by rogerwang** [comment]
It works for me under Ubuntu 19.04:


![ubuntu1904](https://user-images.githubusercontent.com/165401/73119557-a0993800-3f9e-11ea-86ba-b9f07ca50f10.gif)

**Comment by DimPaDev** [comment]
Hello, first of all thanks for looking into this so quickly! Is this a virtual machine ? if yes probably hardware acceleration is off, you can check it out by setting in the console : 

`window.location.href = 'chrome://gpu'`

I just found out that if I set in my machine (not virtual) the chromium argument --disable-gpu and execute the demo the bug disappears but obviously it's not a solution because this way we lose hardware acceleration. 
Also, I noticed that the devtools window must be closed when you test the bug , otherwise (I dont know why) but the bug seems to disappear.

**Comment by rogerwang** [comment]
Yes, it's VirtualBox under macOS, but hardware acceleration is enabled. Could you try changing the window manager there and see if it makes any difference?

**Comment by DimPaDev** [comment]
Ok I will try later the demo with another window manager and let you know, I will also contact later another user too who can try the demo on linux to see if he can confirm the bug or not. On the other hand I found a chromium flag ( --disable-gpu-compositing ) that fixes the bug without disabling entirely the hardware acceleration but obviously it not ideal.

**Comment by DimPaDev** [comment]
@kmare could you please run the demo and see if you have this problem too using the 0.44.0 nightly build?

**Comment by DimPaDev** [comment]
The problem is still there I tested this on KDE and on gnome-shell, I deleted config files too, I created another account and tested it in a virtual machine too with Ubuntu 19.04 but nothing , the stated issue persists.

I found that a possible workaround to bypass the problem (without the chromium flag --disable-gpu-compositing) is this :

```
	var w = nw.Window.get();
	w.on('restore', () => {
		w.resizeBy(+1, 0);
		w.resizeBy(-1, 0);
		console.log('restored!');
	}); 
```

This code works fine on KDE plasma (nw1 and nw2 mode) .

However when you apply this workaround on gnome (for example Ubuntu) the workaround works fine only with nw1 mode, With nw2 mode enabled, the restore event is not getting fired correctly so the workaround fails. @rogerwang should I open another isuue for this gnome NW2-related bug?
